docs: add guideline to setup harvester ui extension in air gapped env (#722)
harvester/docs: docs/airgap.md

sudo -i
docker load -i /tmp/rancher-agent-<version>.tar
```

## Harvester UI extension with Rancher Integration

After Rancher v2.10.x, it's necessary to install Harvester UI extension to access Harvester Dashboard. However, in air gapped environment, installing Harvester UI with network is not available. Please follow below steps to install it in air gapped environment.

> 1. Please pull `rancher/ui-plugin-catalog` image and push it to your private registry.

Reviewer: Is the latest version always the best? And how to determine the latest version.

Author: @a110605 do you have any ideas about this?

Author: Pulling latest version is best. Then, install matched version based on https://docs.harvesterhci.io/v1.5/rancher/harvester-ui-extension/#support-matrix.

Reviewer: I assume the ui-plugin-catalog version is different from the support matrix here. If the latest is the best, should we just mention that the user can check https://hub.docker.com/r/rancher/ui-plugin-catalog/tags ?

Author: I think users can't understand what the version means in https://hub.docker.com/r/rancher/ui-plugin-catalog/tags.

So, I think pulling latest  ui-plugin-catalog version is quick way. After that, when installing harvester, users could choose matched version like this screenshot.

![image](https://github.com/user-attachments/assets/5e88afc5-6c77-4255-941a-57e7f6d10dae)

Reviewer 2: Hi @Yu-Jack , rancher/ui-plugin-catalog doesn't have `latest` tag. We can ask user to pull the `latest (newest)` version of image. 
e.g. 3.4.2 is released in recent days containing ui-ext v1.0.4 build artifacts.
 
https://hub.docker.com/r/rancher/ui-plugin-catalog/tags?name=latest

<img width="728" alt="Screenshot 2025-03-13 at 1 46 20 PM" src="https://github.com/user-attachments/assets/d55cdb2c-3c13-4bed-abc9-7d45beffc2a7" />

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-02.png)

  If your registry requires the username/password, please remember to create secret in `cattle-ui-plugin-system` namespace with `kubernetes.io/dockercfg` or `kubernetes.io/dockerconfigjson` type.

  ```yaml
  apiVersion: v1
  kind: Secret
  metadata:
    name: my-registry-secret-rancher
    namespace: cattle-ui-plugin-system
  type: kubernetes.io/dockerconfigjson
  data:
    .dockerconfigjson: {base64 encoded data}
  ```

4. Wait for few minutes to finish the initilation.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-03.png)

5. After that, there are many different UI extensions you could install.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04.png)

6. Select Harvester to click `install` button to install latest ui extension.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-05.png)

   ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-06.png) 

7. Navigate to `Virtualization Management` screen to import Harvester cluster and access its dashboard.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-07.png)

> ## Trouble Shotting
> 
> ### UI Extensions don't show up
> 
> Please navigate to repository screen to click `Refresh` button

Reviewer 3: ```suggestion
## Troubleshooting

### UI Extensions Do Not Appear

If the **Extensions** screen is empty, go to **Repositories** (**⋮ > Manage Repositories**) and then click **Refresh**.
```

![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04-01.png)

![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04-02.png)

![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04-03.png)

![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04-04.png)

> ### Failed to install UI extensions
>   
> If you run into the failed installation  error message, please check the `uiplugins` resource.
> 
> ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-05-01.png)

Reviewer 3: ```suggestion
### Installation Failed

If you encounter an error during installation, check the `uiplugins` resource.

![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-05-01.png)

Example:
```

```
bash-4.4# k get uiplugins -A
NAMESPACE                 NAME        PLUGIN NAME   VERSION   STATE
cattle-ui-plugin-system   harvester   harvester     1.0.3     pending
bash-4.4# k get uiplugins harvester --namespace cattle-ui-plugin-system -o yaml
apiVersion: catalog.cattle.io/v1
kind: UIPlugin
metadata:
  # skip
  name: harvester
  namespace: cattle-ui-plugin-system
spec:
  plugin:
    endpoint: http://ui-plugin-catalog-svc.cattle-ui-plugin-system:8080/plugin/harvester-1.0.3
```

> Since we use `svc.namespace` as endpoint, please make sure this endpoint is accessible from Rancher. If not, please try to use cluster ip directly like `10.43.33.58:8080/plugin/harvester-1.0.3`.

Reviewer 3: ```suggestion
Ensure that `svc.namespace` is accessible from Rancher. If that endpoint is not accessible, you can directly use a cluster IP such as `10.43.33.58:8080/plugin/harvester-1.0.3`.
```

sudo -i
docker load -i /tmp/rancher-agent-<version>.tar
```

## Harvester UI extension with Rancher Integration

The Harvester UI Extension is required to access the Harvester UI in Rancher v2.10.x and later versions. However, installing the extension over the network is not possible in air-gapped environments, so you must perform the following workaround:

> 1. Please pull `rancher/ui-plugin-catalog:latest` image and push it to your private registry.

Reviewer 3: ```suggestion
1. Pull the image `rancher/ui-plugin-catalog` and then push it to your private registry.
```

Author: @jillian-maroket For this one, I would keep `:latest` word.

Reviewer 3: Agree. Sorry, for some reason I wasn't able to copy that part.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-01.png)

1. Specify the required information.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-02.png)

    - **Catalog Image Reference**: Specify the private registry URL and image repository.
    - **Image Pull Secrets**: Specify the secret used to access the registry when a username and password are required. You must create that secret in the `cattle-ui-plugin-system` namespace. Use either `kubernetes.io/dockercfg` or `kubernetes.io/dockerconfigjson` as the value of `type`.
      Example:

      ```yaml
      apiVersion: v1
      kind: Secret
      metadata:
        name: my-registry-secret-rancher
        namespace: cattle-ui-plugin-system
      type: kubernetes.io/dockerconfigjson
      data:
        .dockerconfigjson: {base64 encoded data}
      ```

1. Click **Load**, and then allow some time for the extension to be loaded.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-03.png)

1. On the **Available** tab, locate the extension named **Harvester**, and then click **Install**.

  ![](/img/v1.5/air-gapped/air-gappted-harvester-ui-extension-04.png)

> 1. Check the [Harvester UI Extension Support Matrix](/v1.5/rancher/harvester-ui-extension/#support-matrix) to install matched version, then click **Install** again. 

Author: new added part

> 1. Check the [Harvester UI Extension Support Matrix](/v1.5/rancher/harvester-ui-extension/#support-matrix) to install matched version, then click **Install** again. 

Reviewer 3: ```suggestion
1. Select the version that matches the Harvester cluster, and then click **Install**.

  For more information, see the [Harvester UI Extension Support Matrix](/v1.5/rancher/harvester-ui-extension/#support-matrix).
```

sudo -i
docker load -i /tmp/rancher-agent-<version>.tar
```

## Harvester UI extension with Rancher Integration

The Harvester UI Extension is required to access the Harvester UI in Rancher v2.10.x and later versions. However, installing the extension over the network is not possible in air-gapped environments, so you must perform the following workaround:

> 1. Pull the image `rancher/ui-plugin-catalog:latest` and then push it to your private registry.

Reviewer: This doesn't work:

```
docker pull rancher/ui-plugin-catalog:latest
Error response from daemon: manifest for rancher/ui-plugin-catalog:latest not found: manifest unknown: manifest unknown
```

Author: Ok, I didn't expect there is no latest tag. Let me revise it.

Reviewer 2: yes. there is no latest tag for rancher/ui-plugin-catalog.

The Harvester UI Extension is required to access the Harvester UI in Rancher v2.10.x and later versions. However, installing the extension over the network is not possible in air-gapped environments, so you must perform the following workaround:

> 1. Pull the image `rancher/ui-plugin-catalog` with [latest tag](https://hub.docker.com/r/rancher/ui-plugin-catalog/tags) and then push it to your private registry.

Reviewer 2: How about ?
```
Pull the image `rancher/ui-plugin-catalog` with the newest tag (https://hub.docker.com/r/rancher/ui-plugin-catalog/tags)
```
